src/components/modals/CategoryModal.tsx

```diff
@@ -1,157 +1,161 @@
+    <TouchableHighlight delayPressIn={60} style={styles.rowContainer} underlayColor={THEME.COLORS.GRAY_4} onPress={() => bridge.resolve(item.raw)}>
+      <View style={styles.rowContent}>
+        <View style={styles.rowCategoryTextWrap}>
+          <FormattedText style={styles.rowCategoryText}>{item.display}</FormattedText>
+        </View>
+        {item.new ? (
+          <View style={styles.rowPlusWrap}>
+            <FormattedText style={styles.rowPlus}>+</FormattedText>
+          </View>
+        ) : null}
+      </View>
+    </TouchableHighlight>
+  ))
 
-    const selectedSubcategories = subCategories.filter(subCategory => {
-      const splittedSubCategory = subCategory.split(':')
-      // @ts-expect-error
-      return splittedSubCategory[0].toLowerCase() === categories[category].syntax.toLowerCase()
-    })
-    const filteredSubcategories = subCategories.filter(subCategory => {
-      const splittedSubCategory = subCategory.split(':')
-      // @ts-expect-error
-      return splittedSubCategory[0].toLowerCase() !== categories[category].syntax.toLowerCase()
-    })
-    return [...selectedSubcategories, ...filteredSubcategories]
-  }
+  return (
+    <AirshipModal bridge={bridge} onCancel={handleCancel}>
+      <TouchableWithoutFeedback onPress={handleCancel}>
+        <View style={styles.airshipContainer}>
+          <FormattedText style={styles.airshipHeader}>{s.strings.transaction_details_category_title}</FormattedText>
+          <View style={styles.inputCategoryMainContainter}>
+            <FormattedText style={styles.inputCategoryListHeader}>{s.strings.tx_detail_picker_title}</FormattedText>
+            <View style={styles.inputCategoryRow}>
+              {categoryOrder.map(item => (
+                <TouchableWithoutFeedback onPress={() => setCategory(item)} key={item}>
+                  <View style={category === item ? styles.inputCategoryContainterSelected : styles.inputCategoryContainter}>
+                    <FormattedText style={styles.inputCategoryText}>{displayCategories[item]}</FormattedText>
+                  </View>
+                </TouchableWithoutFeedback>
+              ))}
+            </View>
+          </View>
+          <View style={styles.inputSubCategoryContainter}>
+            <FormField
+              {...MaterialInputOnWhite}
+              containerStyle={{
+                ...MaterialInputOnWhite.containerStyle,
+                height: THEME.rem(3.44),
+                width: '100%'
+              }}
+              autoFocus
+              returnKeyType="done"
+              autoCapitalize="none"
+              label={s.strings.transaction_details_choose_a_sub_category}
+              fontSize={THEME.rem(0.9)}
+              labelFontSize={THEME.rem(0.65)}
+              onChangeText={setSubcategory}
+              onSubmitEditing={handleSubmit}
+              value={subcategory}
+            />
+          </View>
+          <FlatList
```

Comment:
The contentContainerStyle got removed when this got pulled from TransactionSubcategorySelect.tsx. Do we need that?

Reply:
That style did something a few years back, but now we always pass `0`, so it's not useful anymore.